src/module.ts

```diff
@@ -1474,23 +1482,25 @@ export class Module {
     // Implicitly run costly non-LLVM optimizations on -O3 or -Oz
     if (optimizeLevel >= 3 || shrinkLevel >= 2) optimizeLevel = 4;
 
-    binaryen._BinaryenSetOptimizeLevel(optimizeLevel);
-    binaryen._BinaryenSetShrinkLevel(shrinkLevel);
-    binaryen._BinaryenSetDebugInfo(debugInfo);
+    this.setOptimizeLevel(optimizeLevel);
+    this.setShrinkLevel(shrinkLevel);
+    this.setDebugInfo(debugInfo);
+    this.setAllowHeavyweight(false); // TODO
```

Comment:
```suggestion
    this.setAllowHeavyweight(true); // TODO
```

Reply:
Oh, I was under the impression that the default, i.e. what Binaryen does currently, is not allowing these?

Reply:
No, it's turned off by default. And it will be always turn on due to internally it already applied only for `opt-level == 3` and `shrink-level == 0`

Reply:
Just to clarify: Without changing anything about the output, what would "off" (i.e. false) do that, and turning it "on" (i.e. true) would be the change you want to do on top of it? If so, what do you think about making the functional change a follow-up PR, with this one just introducing the option without changing what's happening currently?

Reply:
Not sure what you mean) `this.setAllowHeavyweight(false)` means the same mode as before and doesn't change anything.

`this.setAllowHeavyweight(true)` is turn on inline mode which allow inline all heavyweight functions even if they exported which improve inline heuristics

Reply:
Yeah, I'm trying to not change anything besides adding the APIs in this PR, and was assuming that you'd go and use the functionality introduced here in a follow-up PR, including updating fixtures. Does that sound good to you?

Reply:
It's just need switch from `false` to `true` in this PR. I think do separate pr for that is unnecessary

Reply:
Alright, will give it a shot once we have a new Binaryen version to see how it affects fixtures.

Reply:
plz switch to `true` (turn on) this mode here

Reply:
I tried locally first but nothing was different, see comment below.

Reply:
yes, this is expected for our test cases